core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreMessageHandler.java

```diff
@@ -59,30 +59,32 @@ private void handleDeleteSubject(DeleteSubjectValue deleteSubjectValue) {
 
         SchemaKey schemaKey = new SchemaKey(subject, version);
         SchemaValue schemaValue = (SchemaValue) this.store.get(schemaKey);
-        schemaValue.setDeleted(true);
-        this.store.put(schemaKey, schemaValue);
-
-
+        if (schemaValue != null) {
+          schemaValue.setDeleted(true);
+          this.store.put(schemaKey, schemaValue);
+        }
       } catch (StoreException e) {
         log.error("Failed to delete subject in the local store");
       }
     }
   }
 
   private void handleSchemaUpdate(SchemaKey schemaKey, SchemaValue schemaObj) {
-    schemaRegistry.guidToSchemaKey.put(schemaObj.getId(), schemaKey);
+    if (schemaObj != null) {
```

Comment:
should we throw an error here instead?

Reply:
nope, we don't want to fail. This is only temporarily handling arresting the bleed. Not many users do this but the most common use case would be health check where getById is not as relevant.

Reply:
I'm looking to create a wholistic solution to handle tombstone messages and delete keys in a future release.